spec/helpers/users_helper_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+require 'cancan/matchers'
+
+RSpec.describe UsersHelper, type: :helper do
+
+  describe "#teams_for?" do
+    let!(:user) { create :user }
+    let!(:invisible_team) { create :team, invisible: true, kind: nil }
+    let(:role_name) { 'student' }
+
+    before do
+      invisible_team.attributes = { roles_attributes: [{ name: role_name, user_id: user.id }] }
```

Comment:
Personal preference, but I find this a bit too verbose...

How about just creating the role?
```ruby
before { create(:role, name: 'student', user: user, team: invisible_team) }
```